frontend/src/pages/dashboard.jsx

```diff
@@ -0,0 +1,339 @@
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  createTheme,
+  ThemeProvider,
+  TextField,
+  MenuItem,
+  Select,
+  InputLabel,
+  Button,
+  IconButton,
+  Dialog,
+  DialogTitle,
+  Box,
+  Typography,
+  Fab,
+} from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import AddIcon from "@mui/icons-material/Add";
+import {
+  addExpense,
+  deleteExpense,
+  fetchExpenses,
+  updateExpense,
+} from "../services/api";
+import { useEffect, useState } from "react";
+
+const theme = createTheme({
+  typography: {
+    fontFamily: "Segoe UI",
+  },
+  palette: {
+    primary: {
+      main: "#ffbd59", // orange
+      contrastText: "#080357", // dark blue
+    },
+  },
+});
+
+const Dashboard = () => {
+  const [expenses, setExpenses] = useState([]);
+  const [editRowId, setEditRowId] = useState(null);
+  const [category, setCategory] = useState("");
+  const [amount, setAmount] = useState(0);
+  const [description, setDescription] = useState("");
+  const [newCategory, setNewCategory] = useState("");
+  const [newAmount, setNewAmount] = useState(0);
+  const [newDescription, setNewDescription] = useState("");
+  const [openDeleteDialogue, setOpenDeleteDialogue] = useState(false);
+  const [openAddDialogue, setOpenAddDialogue] = useState(false);
+  const [selectedExpense, setSelectedExpense] = useState({});
+  const [displayMessage, setDisplayMessage] = useState("");
+
+  useEffect(() => {
+    const getExpenses = async () => {
+      try {
+        const response = await fetchExpenses();
+        setExpenses(response);
```

Comment:
There is an issue with how you are setting your state for the edit function. For example, when I try to go edit something, it is showing me the state of the previous item rather than the item in question. 

Hint: make sure you are updating your state properly for the given functions.

Reply:
Got it .